File: hw/pazmino/u3/HW37StrategyExercise/HW38StrategySortApp/src/main/java/ec/edu/espe/hw38strategysortapp/controller/SortingContext.java
package ec.edu.espe.hw38strategysortapp.controller;

import ec.edu.espe.hw38strategysortapp.model.BubbleSort;
import ec.edu.espe.hw38strategysortapp.model.InsertionSort;
import ec.edu.espe.hw38strategysortapp.model.QuickSort;
import ec.edu.espe.hw38strategysortapp.model.SortingStrategy;

/**
 *
 * @author Brandon Pazmino
 */
public class SortingContext {

    private SortingStrategy ss;

    public int[] sort(int data[]) {
        int size = data.length;
        ss = setSortStrategy(size);
        return ss.sort(data);
    }

    //
    public SortingStrategy setSortStrategy(int n) {
        if (n > 0 && n < 30) {
            ss = new BubbleSort();
        }
        if (n >= 30 && n < 100) {
            ss = new InsertionSort();
        }
        if (n >= 100) {
            ss = new QuickSort();
        }
        return ss;
    }

}

File: EXAM/pazmino/u3/PazmiñoStrategy/controller/sorting_context.py

from model.bubble_sort import BubbleSort
from model.insertion_sort import InsertionSort
from model.quick_sort import QuickSort

class SortingContext:
    def __init__(self):
        self.sort_strategy = None

    def sort(self, data):
        size = len(data)
        self.set_sort_strategy(size)
        return self.sort_strategy.sort(data)

    def set_sort_strategy(self, n):
        if n > 1 and n < 6:
            self.sort_strategy = BubbleSort()
        elif n >= 6 and n < 11:
            self.sort_strategy = InsertionSort()
        elif n >= 11:
            self.sort_strategy = QuickSort()

